src/main/java/net/dries007/tfc/common/blocks/LogPileBlock.java

```diff
@@ -0,0 +1,129 @@
+import net.minecraft.item.BlockItemUseContext;
+import net.minecraft.item.ItemStack;
+import net.minecraft.state.EnumProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.tags.BlockTags;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Direction;
+import net.minecraft.util.Hand;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.network.NetworkHooks;
+
+import net.dries007.tfc.common.TFCTags;
+import net.dries007.tfc.common.tileentity.InventoryTileEntity;
+import net.dries007.tfc.common.tileentity.LogPileTileEntity;
+import net.dries007.tfc.util.Helpers;
+
+public class LogPileBlock extends Block implements IForgeBlockProperties
+{
+    public static final EnumProperty<Direction.Axis> AXIS = BlockStateProperties.HORIZONTAL_AXIS;
+    private final ForgeBlockProperties properties;
+
+    public LogPileBlock(ForgeBlockProperties properties)
+    {
+        super(properties.properties());
+        this.properties = properties;
+        registerDefaultState(getStateDefinition().any().setValue(AXIS, Direction.Axis.X));
+    }
+
+    @Override
+    public ForgeBlockProperties getForgeProperties()
+    {
+        return properties;
+    }
+
+    @Override
+    public BlockState getStateForPlacement(BlockItemUseContext context)
+    {
+        return defaultBlockState().setValue(AXIS, context.getHorizontalDirection().getAxis());
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateContainer.Builder<Block, BlockState> builder)
+    {
+        super.createBlockStateDefinition(builder.add(AXIS));
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public BlockState updateShape(BlockState stateIn, Direction facing, BlockState facingState, IWorld worldIn, BlockPos currentPos, BlockPos facingPos)
+    {
+        if (!worldIn.isClientSide() && worldIn instanceof World)
+        {
+            if (facingState.is(BlockTags.FIRE))
+            {
+                BurningLogPileBlock.tryLightLogPile((World) worldIn, currentPos);
+            }
+        }
+        return super.updateShape(stateIn, facing, facingState, worldIn, currentPos, facingPos);
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public void onRemove(BlockState state, World world, BlockPos pos, BlockState newState, boolean isMoving)
+    {
+        InventoryTileEntity te = Helpers.getTileEntity(world, pos, InventoryTileEntity.class);
+        if (state.hasTileEntity() && (!state.is(newState.getBlock()) || !newState.hasTileEntity()))
+        {
+            if (te != null)
+                te.onBreak();
+            world.removeBlockEntity(pos);
+        }
+    }
```

Comment:
There has to be a better way of handling this than copy pasting the same vanilla method across every single TE, and not delegating to super at all.

Reply:
This was actually a big bug that I was chasing a while. The problem is essentially that you only want to drop the inventory when you've actually broken the block (not when it's being manipulated in some way). This is the most natural place for that.

Other option is to call te.Remove() a bunch of times / clear the inventory a lot

Reply:
Yeah, but we can still avoid the copy-paste + fix the lack of super call.
- Add `class DeviceBlock extends Block`, as for most inventory tile entity blocks they will just be extending `Block` rather than basing off vanilla functionality. Override the method there, but,
  - Call `super` rather than removing the block entity yourself
  - The only necessary check there is the `state.is(newState.getBlock())` for our intents and purposes
  - Delegate to `InventoryTileEntity#onRemove` (renamed from `onBreak` as it's a passthrough)
- Throw the basic `IForgeProperties` / `ForgeBlockProperties` impl. onto `DeviceBlock`